File: exercice3/main.ts
class User {
  firstName: string;
  lastName: string;
  birthDate: string;

  constructor(firstName: string, lastName: string, birthDate?: string) {
    this.firstName = firstName;
    this.lastName = lastName;
    if (birthDate) {
      this.birthDate = birthDate;
    }
  }

  displayInformation() {
    if (!this.birthDate) {
      console.log("Je m'appelle " + this.firstName + " " + this.lastName)
    } else {
      console.log("Je m'appelle " + this.firstName + " " + this.lastName + " je suis né le " + this.birthDate);
    }
  }
}

let mario: User = new User("Mario", "DeVito");

mario.displayInformation();

File: exercice3/main.java
class Playground {
    public static void main(String[ ] args) {
        User mario = new User("Mario", "DeVito", "12-03-1983");
        User luigi = new User ("Luigi", "DeVito");
        System.out.println("Je m'appelle " + mario.firstName + " " + mario.lastName + " et je suis né le " + mario.birthDate);
    }
}

public class User {
    String firstName;
    String lastName;
    String birthDate;

    public User(String firstName, String lastName, String birthDate){
        this.firstName = firstName;
        this.lastName = lastName;
        this.birthDate = birthDate;
    }

    public User(String firstName, String lastName){
        this.firstName = firstName;
        this.lastName = lastName;
    }
}
